Jin/ChatTCPServer/src/threadex/ServerThread.java

```diff
@@ -0,0 +1,260 @@
+				List<String> temp1 = new ArrayList<String>();
+
+				if (cr.getRoomName().equals(roomName)) {
+					temp1 = cr.getUserlist();
+					if (temp1.contains(id)) {
+						temp1.remove(id);
+						cr.setUserlist(temp1);
+						chatList.set(chatList.indexOf(cr), cr);
+					}
+				}
+			}
+
+			Iterator<ChatRoom> iter = chatList.iterator();
+			while (iter.hasNext()) {
+				ChatRoom cr = iter.next();
+				if (cr.getUserlist().isEmpty() || cr.getUserlist().size() == 0) {
+					iter.remove();
+				}
+			}
+
+			break;
+		}
+
+		return result;
+	}
+
+	@Override
+	public void run() {
+		super.run();
+
+		try {
+
+			while (true) {
+
+				// 수신(recieve)
+
+				BufferedReader reader = new BufferedReader(
+						new InputStreamReader(socket.getInputStream()));
+				String str = reader.readLine();
+
+				System.out.println("client로부터 받은 메시지:" + str);
+
+				String sendStr = readRecieve(str);
+
+				// 송신(send)
+
+				for (Socket s : list) {
+
+					PrintWriter writer = new PrintWriter(s.getOutputStream());
+					writer.println(sendStr);
+					writer.flush();
+
+				}
+
+				Thread.sleep(300);
+
+				if (protocol == INIT) {
+
+					if (!userIdList.isEmpty()) {
+
+						String initUser = protocol + "|";
+
+						for (String s : userIdList) {
+							initUser += s + ",";
+						}
+
+						initUser = initUser.substring(0, initUser.length());
+
+						for (Socket s : list) {
+							if (socket == s) {
+								PrintWriter writer = new PrintWriter(s.getOutputStream());
+
+								writer.println(initUser);
+								writer.flush();
+								System.out.println("initUser : " + initUser);
+								break;
+							}
+
+						}
+					}
+
+					Thread.sleep(300);
+
+					if (!chatList.isEmpty()) {
+						protocol = INITCHATROOM;
+
+						for (ChatRoom cr : chatList) {
+
+							Thread.sleep(300);
+
+							String initChatResult = protocol + "|" + cr.getRoomName()
+									+ "|";
+
+							for (String s : cr.getUserlist()) {
+								initChatResult += s + ",";
+							}
+							for (Socket s : list) {
+								if (socket == s) {
+									PrintWriter writer = new PrintWriter(
+											s.getOutputStream());
+
+									writer.println(initChatResult);
+									writer.flush();
+									System.out.println(
+											"initChatResult : " + initChatResult);
+									break;
+								}
+								Thread.sleep(300);
+
+							}
+							Thread.sleep(300);
+						}
+					}
+				}
+
+				Thread.sleep(300);
+
+			}
+		} catch (
+
+		SocketException e) {
+			// SocketException
+			System.out.println("연결이 끊긴 IP:" + socket.getInetAddress());
+			list.remove(socket);
+
+			// 접속되어 있는 남아 있는 클라이언트 출력
+			for (Socket s : list) {
+				System.out.println(
+						"접속되어 있는 IP: " + s.getInetAddress() + " Post: " + s.getPort());
+			}
+			try {
+				socket.close();
+			} catch (IOException e1) {
+				// TODO Auto-generated catch block
+				e1.printStackTrace();
+			}
+		} catch (IOException e) {
+			e.printStackTrace();
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+		}
```

Comment:
따로 처리를 안해준다면 아래처럼 합쳐버려도 괜찮을 것 같아요
```suggestion
		} catch (Exception e) {
			e.printStackTrace();
		}
```